addons/mrp_byproduct/models/mrp_production.py

```diff
@@ -39,6 +39,9 @@ def _generate_moves(self):
         """
         res = super(MrpProduction, self)._generate_moves()
         for production in self.filtered(lambda production: production.bom_id):
-            for sub_product in production.bom_id.sub_products:
-                production._create_byproduct_move(sub_product)
+            factor = production.product_uom_id._compute_quantity(production.product_qty, production.bom_id.product_uom_id) / production.bom_id.product_qty
+            boms, lines = production.bom_id.explode(production.product_id, factor, picking_type=production.bom_id.picking_type_id)
+            for (bom, bom_data) in boms:
+                for sub_product in bom.sub_products:
+                    production._create_byproduct_move(sub_product)
```

Comment:
I would say this is more extra features.  Maybe also this should be handled by the explode itself.  Is there no way to achieve this by only putting the byproducts in the main bom and reusing the same routing (if you use the same routing in subboms, it won't duplicate the wos)?

Reply:
Thank you for your review!

I agree, it is a feature, as it changes behaviour. So definitly not stable.
The expode resides in the `mrp` module, while the byproduct is nown only when installing the byproduct module... I think strategically, the ideal would be to merge `mrp_byproduct` and `mrp`. But that's out of scope.

It would be achievable, but the use case is like 150 different presentations out of a single formula with some decent amount of different formulas... It would just become un-manageable after a while...
What would be your ides to include it into the explode? a good old `hasattr()` check?